dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/SAMLHelper.java

```diff
@@ -0,0 +1,468 @@
+        role.setRoleKey(roleKey);
+        role.setEditUsers(true);
+        role.setEditPermissions(false);
+        role.setEditLayouts(false);
+        role.setDescription("");
+        role.setId(UUIDGenerator.generateUuid());
+
+        // Setting SYSTEM role as a parent
+        role.setSystem(isSystem);
+        Role parentRole = roleAPI.loadRoleByKey(Role.SYSTEM);
+        role.setParent(parentRole.getId());
+
+        String date = DateUtil.getCurrentDate();
+
+        ActivityLogger.logInfo(ActivityLogger.class, getClass() + " - Adding Role",
+                "Date: " + date + "; " + "Role:" + roleKey);
+        AdminLogger.log(AdminLogger.class, getClass() + " - Adding Role", "Date: " + date + "; " + "Role:" + roleKey);
+
+        try {
+            role = roleAPI.save(role, role.getId());
+        } catch (DotDataException | DotStateException e) {
+            ActivityLogger.logInfo(ActivityLogger.class, getClass() + " - Error adding Role",
+                    "Date: " + date + ";  " + "Role:" + roleKey);
+            AdminLogger.log(AdminLogger.class, getClass() + " - Error adding Role",
+                    "Date: " + date + ";  " + "Role:" + roleKey);
+            throw e;
+        }
+
+        return role;
+    }
+
+    private String toString(final String... rolePatterns) {
+        return null == rolePatterns ? DotSamlConstants.NULL : Arrays.asList(rolePatterns).toString();
+    }
+
+    protected User createNewUser(final User systemUser, final Attributes attributesBean,
+                                 final IdentityProviderConfiguration identityProviderConfiguration) {
+        User user = null;
+
+        try {
+
+            final String nameID = this.samlAuthenticationService.getValue(attributesBean.getNameID());
+            try {
+
+                user = this.userAPI.createUser(nameID, attributesBean.getEmail());
+            } catch (DuplicateUserException due) {
+
+                user = this.onDuplicateUser(attributesBean, identityProviderConfiguration, nameID);
+            }
+
+            user.setFirstName(attributesBean.getFirstName());
+            user.setLastName(attributesBean.getLastName());
+            user.setActive(true);
+
+            user.setCreateDate(new Date());
+            user.setPassword(PublicEncryptionFactory.digestString(UUIDGenerator.generateUuid() + "/" + UUIDGenerator.generateUuid()));
+            user.setPasswordEncrypted(true);
+
+            this.userAPI.save(user, systemUser, false);
+            Logger.debug(this, ()-> "User with NameID '" + nameID + "' and email '" +
+                    attributesBean.getEmail() + "' has been created.");
+
+        } catch (Exception e) {
+
+            final String errorMsg = "Error creating user with NameID '" + this.samlAuthenticationService.getValue(attributesBean.getNameID()) + "': " +
+                    "" + e.getMessage();
+            Logger.error(this, errorMsg, e);
+            throw new SamlException(errorMsg);
```

Comment:
Codacy found an issue: [New exception is thrown in catch block, original stack trace may be lost](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)